modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt

```diff
@@ -0,0 +1,114 @@
+package au.com.shiftyjelly.pocketcasts.player.view.transcripts
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+private const val SEARCH_DEBOUNCE = 300L
+
+@HiltViewModel
+class TranscriptSearchViewModel @Inject constructor(
+    private val kmpSearch: KMPSearch,
+) : ViewModel() {
+    private var _searchSourceText: String = ""
+
+    private val _searchQueryFlow = MutableStateFlow("")
+    val searchQueryFlow = _searchQueryFlow.asStateFlow()
+
+    private val _searchState = MutableStateFlow(SearchUiState())
+    val searchState = _searchState.asStateFlow()
+
+    init {
+        @OptIn(FlowPreview::class)
+        _searchQueryFlow
+            .debounce(SEARCH_DEBOUNCE)
+            .onEach { searchQuery ->
+                performSearch(searchQuery)
+            }
+            .launchIn(viewModelScope)
+    }
+
+    fun setSearchSourceText(searchSourceText: String) {
+        resetSearch()
+        this._searchSourceText = searchSourceText
+    }
+
+    fun onSearchQueryChanged(searchQuery: String) {
+        _searchQueryFlow.value = searchQuery
+    }
+
+    private fun performSearch(searchTerm: String) {
+        viewModelScope.launch {
```

Comment:
Looking at the code now it would be better to make this a suspending function that uses a correct dispatcher. It is used only in `.onEach()` on the `_searchQueryFlow`.

Reply:
I made it suspending,  using a default dispatcher in 60764be.